crates/subspace-networking/src/node_runner.rs

```diff
@@ -135,9 +154,30 @@ impl NodeRunner {
     }
 
     async fn handle_peer_dialing(&mut self) {
-        let connected_peers = self.swarm.connected_peers().cloned().collect::<Vec<_>>();
         let local_peer_id = *self.swarm.local_peer_id();
+        let connected_peers = self.swarm.connected_peers().cloned().collect::<Vec<_>>();
+
+        // Handle reserved peers first.
+        if !self.reserved_peers.is_empty() {
+            trace!(%local_peer_id, "Checking reserved peers connection: {:?}", self.reserved_peers);
+
+            let reserved_peers = convert_multiaddresses(self.reserved_peers.clone())
+                .into_iter()
+                .collect::<HashMap<_, _>>();
```

Comment:
Maybe worth storing them in the hashmap then?

Reply:
I try to evaluate it within the next task: https://github.com/subspace/subspace/issues/770